### PR TITLE
Suggesting patches on restructuring the code

### PR DESCRIPTION
**Pull request by vishnuchalla**
#### Description
Suggesting patches on top of the existing code.
Still there is lot of code that needs to be changed. Will take them in incremental PRs.

#### Testing
Tested with `CLI`. Verified both `stdout` and `csv` results.

### CODE REVIEW

File: pkg/results/result.go [hunk 1]
```diff
@@ -101,6 +104,17 @@ func TCPThroughputDiff(s ScenarioResults) (float64, error) {
 	return calDiff(hostPerf, podPerf), nil
 }
 
+// Method to init common table structure.
+func initTable(header []string) (*tablewriter.Table) {
```
Comment (jtaleric) [inline pkg/results/result.go:108]: Possible bad rebase?

**Review by jtaleric — CHANGES_REQUESTED** [review]
This patchset seems out of sync with main.

File: pkg/archive/archive.go [hunk 2]
```diff
@@ -113,6 +113,66 @@ func IndexDocs(client *opensearch.Client, docs []Doc) error {
 	return nil
 }
 
+// Common csv header fields.
+func commonCsvHeaderFields() []string {
```
[the remark this answers is not in the document]
Comment (vishnuchalla) [reply pkg/archive/archive.go:123]: Added Driver field and rebased on top of main. Thanks!

**Comment by jtaleric** [comment]
So, testing this patch i see 
```shell
15:26 $ cat result-1680716211.csv
Driver,Profile,Same node,Host Network,Service,Duration,Parallelism,# of Samples,Message Size,Avg Throughput,Throughput Metric,99%tile Observed Latency,Latency Metric
netperf,TCP_STREAM,false,false,false,10,1,3,1024,4923.653333,Mb/s,4,usec
iperf3,TCP_STREAM,false,false,false,10,1,3,1024,3111.568725,Mb/s,0,usec
netperf,TCP_STREAM,false,false,false,10,2,3,1024,9235.816667,Mb/s,9,usec
iperf3,TCP_STREAM,false,false,false,10,2,3,1024,3828.860501,Mb/s,0,usec
netperf,TCP_RR,false,false,false,10,2,3,1024,9846.400000,OP/s,244.66666666666666,usec
,,false,false,false,0,0,0,0,NaN,,NaN,usec
```

using 
```shell
15:27 $ cat test.yaml 
---
TCPStream:
   parallelism: 1
   profile: "TCP_STREAM"
   duration: 10
   samples: 3
   messagesize: 1024

TCPStreami2:
   parallelism: 2
   profile: "TCP_STREAM"
   duration: 10
   samples: 3
   messagesize: 1024

TCPRR2:
   parallelism: 2
   profile: "TCP_RR"
   duration: 10
   samples: 3
   messagesize: 1024
```

This might have been a problem with the introduction of iperf.

**Comment by vishnuchalla** [comment]
> > So, testing this patch i see
> > ```shell
> > 15:26 $ cat result-1680716211.csv
> > Driver,Profile,Same node,Host Network,Service,Duration,Parallelism,# of Samples,Message Size,Avg Throughput,Throughput Metric,99%tile Observed Latency,Latency Metric
> > netperf,TCP_STREAM,false,false,false,10,1,3,1024,4923.653333,Mb/s,4,usec
> > iperf3,TCP_STREAM,false,false,false,10,1,3,1024,3111.568725,Mb/s,0,usec
> > netperf,TCP_STREAM,false,false,false,10,2,3,1024,9235.816667,Mb/s,9,usec
> > iperf3,TCP_STREAM,false,false,false,10,2,3,1024,3828.860501,Mb/s,0,usec
> > netperf,TCP_RR,false,false,false,10,2,3,1024,9846.400000,OP/s,244.66666666666666,usec
> > ,,false,false,false,0,0,0,0,NaN,,NaN,usec
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > using
> > ```shell
> > 15:27 $ cat test.yaml 
> > ---
> > TCPStream:
> >    parallelism: 1
> >    profile: "TCP_STREAM"
> >    duration: 10
> >    samples: 3
> >    messagesize: 1024
> > 
> > TCPStreami2:
> >    parallelism: 2
> >    profile: "TCP_STREAM"
> >    duration: 10
> >    samples: 3
> >    messagesize: 1024
> > 
> > TCPRR2:
> >    parallelism: 2
> >    profile: "TCP_RR"
> >    duration: 10
> >    samples: 3
> >    messagesize: 1024
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This might have been a problem with the introduction of iperf.
> 
> This patch did not introduce this issue, but I am wondering if we can fix this issue in this patchset, since it is making changes to the `archive` package.
> 
> https://github.com/jtaleric/k8s-netperf/blob/main/pkg/archive/archive.go#L291 in this loop, we need to check if https://github.com/jtaleric/k8s-netperf/blob/main/pkg/archive/archive.go#L296 `profile` contains "RR" and driver is iperf3, we just need to conintue.

I don't think that would be the correct check. Because when I log and see the row we are getting empty strings in case of those incorrect rows being written into csv. 

```
%v {{0 0  0 0 false }   false false { } { } {0 0 0  } 0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC false [] [] {0 0 0 0 0 0 0 0} {0 0 0 0 0 0 0 0} {[]} {[]}}
,,false,false,false,0,0,0,0,NaN,,NaN,usec
```

I think we might have to check for row.Profile/row.Driver with empty string values and eliminate those rows or else populate them with the data which you are expecting. Let me know your thoughts on this. Thanks!

**Comment by jtaleric** [comment]
Addressing the above comments in #61

**Review by jtaleric — APPROVED** [review]
lgtm!

**Pull request merged**